src/include/interval_set.h

```diff
@@ -764,6 +764,22 @@ class interval_set {
     return std::move(m);
   }
 
+  /*
+  * Round down interval starts and round up interval ends alignment the specified @alignment
+  */
+  void align(T alignment) {
+    interval_set<T> tmp;
+    for (const auto& [start, len] : m) {
+      T aligned_start = (start / alignment) * alignment;
+      T aligned_len = ((start + len + alignment - 1) / alignment) * alignment - aligned_start;
+
+      if (!tmp.contains(aligned_start, aligned_len)) {
+	tmp.insert(aligned_start, aligned_len);
```

Comment:
It was actually written before the optimisations and the "contains" was an attempt to avoid a double insert, although looking again I can see that this would not always work.  With the new insert code, I think we can do away with the contains() call completely.

Reply:
Pre-formatted version of Bill's example:

    For example a set with 4~1 and 8~10 that is then aligned to 10 will try and insert aligned intervals 0~10 and 0~20